### ISSUE TITLE
What is the relation between ADFS and Azure AD?

### ISSUE THREAD

**Issue by mchudinov** [aspnet/Docs#8217]
It is unclear from the article what is the relation between ADFS and Azure AD.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 59d6d442-8d46-3f9e-e75a-ad9d900fc117
* Version Independent ID: d07c4058-452f-cdac-c59d-8f09e70a43f3
* Content: [Authenticate users with WS-Federation in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/ws-federation?view=aspnetcore-2.1)
* Content Source: [aspnetcore/security/authentication/ws-federation.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/security/authentication/ws-federation.md)
* Product: **aspnet-core**
* GitHub Login: @chlowell
* Microsoft Alias: **scaddie**

**Comment by CamSoper** [comment]
Sure, that's an easy one to get confused on.

@mchudinov So basically ADFS is a Security Token Service layer on top of Active Directory. So say you've got web apps, and you want the web apps to use your domain controller (DC) for authentication but the users aren't actually connected to your corporate network (think remote workers or B2B scenarios).

Azure AD, on the other hand, isn't just an STS layer - It's also the entire directory. It's not the same as your on-premises DC, as there are some subtle differences. Most importantly Azure AD has the ability to run completely independent of a traditional Windows domain all on its own.  It's all hosted in the cloud, directory *and* STS.  Though it *can* be run independent of a DC, they generally are [kept in sync](https://docs.microsoft.com/en-us/azure/active-directory/connect/active-directory-aadconnect).

Why would you choose one over the other? Lots of reasons. Azure AD is required for Office 365, for starters. Azure AD is pretty easy to configure and scales automatically. ADFS requires more work to manage, but arguably provides more customization options.  It's not unusual to see an org running both because they can be federated.

Basically, the take-away here is that they are both STSs and an org can rely on either or both, depending on needs.

**Comment by Rick-Anderson** [comment]
@CamSoper  that's really great info. Should it be added to the doc?

**Comment by CamSoper** [comment]
@Rick-Anderson I was looking at the doc a bit. I don't know that this explanation belongs in this doc. I can see why it's confusing, though. The first part shows how to setup the example in ADFS, and then it **also** shows how to set up the example in Azure AD.  Then it goes on to do the rest with the assumption you did one or the other in the first part.

IMHO the best option to reduce confusion is to either split them out into separate docs (one for ADFS and one for Azure AD) or put them on different tabs or, at a minimum, clarify that we're dealing with two different use cases.

**Comment by mchudinov** [comment]
I know what is the difference between ADFS and Azure AD.
What was unclear that we choose between them in this example. Not using both one behind another.
